activation_key_logger.go

```diff
@@ -0,0 +1,190 @@
+package micrologger
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	KeyLevel     = "level"
+	KeyVerbosity = "verbosity"
+)
```

Comment:
These are two reserved keys we want to treat in a special way when the activation key logger is used. If not, these keys do not mean anything special.

Reply:
Do you think we need both mechanisms? As a user I have no clear indication how to combine they together. I think they both serve exact same purpose, therefore I would leave only one of them.

I would be more keen to leave only verbosity. We made unfortunate decision to log messages like `"info": "..."`. Having messages like `"level": "debug", "info": "..."` or even ``"level": "info", "info": "..."` is confusing.

Reply:
A log level is something completely different to log verbosity. They are both categories but have different semantics. A log level classifies logs emitted based on their occasion at runtime. They for instance indicate that there was a situation that is informational but also near a critical point towards an error case, so the log level `warning` would be used. 

When you want to debug something you classify logs based on their verbosity. You can say that all user facing code should produce debug logs with verbosity 1. You can then say verbosity 2 is used for network related logs and so on. 

These are just very common and flexible parameters and the user is free to use them as they see fit. When you do not have a clear indication how or even if to use these at all, then you do not have to use them. This proposal here though is for users who want to use them for the reasons they find appropriate. I for instance find these appropriate in `opsctl` as stated in the PR description.

Reply:
> I for instance find these appropriate in opsctl as stated in the PR description

I would like to know how they would be applied instead of where.

In `glog` terms verbosity and level are the same and I find it intuitive.
Mapping verbosity `2` with `network` related logs sounds like pure mystery. `network` looks more like module (also in `glog` terms).

You can accuse me of being fussed about `glog`, which I'm not. But I think following interfaces which are already widely known is a good thing.

I can disagree and commit.

Reply:
Instead of writing this to logging library, could it be useful to create a separate tool where you could define these criterias and it would parse log from STDIN and remove JSON cruft and write filtered log to STDOUT (or optionally pipe JSON out)?

Reply:
You mean something like that https://github.com/giantswarm/luigi ?

Reply:
That does not work because there are commands like `opsctl create kubeconfig` where there is valid output I do not want to filter. I just want to have logging. I want to have debug logging. And I want to have debug logging for different depths on insights. It is just a multidimensional filter and when people are happy with their one dimensional filters that is fair. I just wanted to provide something more useful.